### ISSUE TITLE
No log file produced after a failure

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#705]
In GitLab by @acarreri on Jun 3, 2021, 14:12

Hi @eferre1 , @mcastril ,  (sorry, again not sure to tag the correct people)

The first SIM job of my experiment, `a3qa` failed (received the notification and confirmed by the GUI) but no log file has been saved in the `/esarchive/autosubmit/a3qa/tmp/LOG_a3qa/` folder. But it is in MN4.

I don't think it's a normal behaviour. Could you have a look at it please?

Thanks

Aude

**Comment by kinow** [comment]
In GitLab by @eferre1 on Jun 3, 2021, 14:39

Hi @acarreri, not having the log in `/esarchive/autosubmit/a3qa/tmp/LOG_a3qa/` is an issue of autosubmit that is caused due to a problem with the AS machine retrieving the files. This is not related to the `SIM` failure, so I think that @dbeltrankyl or @wuruchi will be the ones who can help you better in case that is not only a temporal problem.

About the `SIM` failure, I'll take a look into it if you need it, to try to understand what caused it.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Jun 3, 2021, 14:46

Hello @acarreri 

In the log of the last run of your experiment, I found:

```
2021-06-03 12:12:40,544 Job a3qa_19800101_fc00_1_SIM inside wrapper a3qa_ASThread_16226243077720_1248_3 is running for longer than it's wallclock!
2021-06-03 12:12:40,544 Job a3qa_19800101_fc00_1_SIM is FAILED
```

It seems that the SIM job was canceled by the platform. I am not sure about the absence of the log though.

**Comment by kinow** [comment]
In GitLab by @acarreri on Jun 3, 2021, 15:03

Thanks @eferre1 and sorry for having tagged you in a random way.

No worries for the failure, I know from where it comes. A sym link is missing in the forcing data, even if I don't understand how it could happen. I fixed that, will relaunch the SIM job.

**Comment by kinow** [comment]
In GitLab by @acarreri on Jun 3, 2021, 15:05

Yes, it's another weird case when the job didn't work since almost the beginning but didn't fail at that time, waiting for the time limit. I think it's a known issue. 

But I think it should save the log file in esarchive even in this case, it doesn't seem as a normal behaviour.

**Comment by kinow** [comment]
In GitLab by @eferre1 on Jun 3, 2021, 15:07

It's ok, don't worry, this is not a usual error and it isn't easy to know that is related to autosubmit if you haven't seen it before. I'm happy to being able to help :smiley:.

**Comment by kinow** [comment]
In GitLab by @acarreri on Jun 3, 2021, 15:24

It's probably linked to the wrapper, no?
I have to kill it manually in mn4 to be able to relaunch the SIM job. (maybe a `recovery` would have done the job here).

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 3, 2021, 16:24

Hi @acarreri. In fact, looking at the log, Autosubmit detected the wallclock issue and set the job as FAILED, so I think it should retrieve the logs:

```
2021-06-03 12:12:40,543 Job a3qa_19800101_fc00_1_SIM is RUNNING
2021-06-03 12:12:40,544 Job a3qa_19800101_fc00_1_SIM inside wrapper a3qa_ASThread_16226243077720_1248_3 is running for longer than it's wallclock!
2021-06-03 12:12:40,544 Job a3qa_19800101_fc00_1_SIM is FAILED
2021-06-03 12:12:40,962 Checking jobs for platform=transfer_node
2021-06-03 12:12:40,965 Checking jobs for platform=nord3
```

Then I think that it should cancel the wrapper (we are trying to modify this behaviour), but in fact it didn't do it. It seemed that it was on a kind of deadlock for a while, because three hours later it still was checking the wrapper status:

```
2021-06-03 15:08:21,008 Checking jobs for platform=local
2021-06-03 15:08:21,010 Checking jobs for platform=marenostrum4
2021-06-03 15:08:21,011 Checking Wrapper 16138707
2021-06-03 15:08:23,932 Checking inner jobs status
2021-06-03 15:08:23,936 Checking jobs for platform=transfer_node
2021-06-03 15:08:23,938 Checking jobs for platform=nord3
```

When did you cancel the wrapper job?

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 3, 2021, 16:25

@acarreri you did it right, @eferre1 should be the one supporting you whenever it's an Auto-EC-Earth problem or you are not sure if it' about Auto-EC-Earth or Autosubmit.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 7, 2021, 15:17

Posting @acarreri comment:

> I canceled the wrapper job a bit after posting my previous post (so after the second check that you posted in your comment). And I did a `recovery` also.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 7, 2021, 15:21

I'll close the issue because it was a system issue and there was no log to retrieve.

**Comment by kinow** [comment]
In GitLab by @acarreri on Jun 30, 2021, 14:33

Hi guys,

I'm sorry, I re-opening this issue, because it happened again with another experiment, `a3rp`.

So indeed, there are "special" experiments because ocean-only with some errors that don't make the SIM job fail during the execution but wait for the wallclock to fail. 

However, I could expect the log file to be saved in esarchive after the wallclock is reached, right? But it's not saved on esarchive. And actually, there is only a `a3rp_19590101_fc00_2_SIM.cmd.err` in mn4 (`/gpfs/scratch/bsc32/bsc32283/a3rp/LOG_a3rp`).

I don't understand well what happens, but I think something's wrong with the log files in this case. And I don't think it's happening without the wrappers (not totally certain though).

Could you have a look at it please?

Many thanks

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jun 30, 2021, 14:37

Hello,

We had issues this morning (noticed during the hands on) regarding to esarchive so I think what happened is that the log retrial mechanism expired the maximum retrials because it was failing for a long period of time

**Comment by kinow** [comment]
In GitLab by @acarreri on Jun 30, 2021, 14:51

Well, if it's because of that, exactly the same problem with esarchive happened for the previous experiment mentioned in the issue.

Which could be considered as strange random errors, because it's really the same kind of "special" experiments, with wrappers and in both cases, no log files are produced for the corresponding SIM job, no even in mn4 (cf this [comment](https://earth.bsc.es/gitlab/es/autosubmit/-/issues/705#note_137485) for the previous experiment in this issue).

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 30, 2021, 16:56

@dbeltrankyl can you confirm that we retrieve the error log files when the wrapper is canceled by AS?

**Comment by kinow** [comment]
In GitLab by @mcastril on Jul 5, 2021, 15:30

@dbeltrankyl confirms that the fact that it was canceled due to wallclock should not have any impact.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jul 12, 2021, 15:45

@acarreri we are talking about this problem right now.

You said that the first chunk in the wrapper was failing, but that there is only an `a3rp_19590101_fc00_2_SIM.cmd.err` file. Wasn't any logfile for the first chunk? Or is the 2nd the one that failed in the second case?

**Comment by kinow** [comment]
In GitLab by @acarreri on Jul 12, 2021, 17:08

Hi @mcastril ,

Sorry, I probably missed the "right now"!

For `a3rp`, the first SIM ran correctly. But not the second SIM, the failure was related to the surface_restoring files. It's an error that doesn't make the job failed, but just waiting to the wallclock limit. 

And for the failure of this second SIM, I never got the log file.

**Comment by kinow** [comment]
In GitLab by @acarreri on Aug 11, 2021, 14:17

Hi all,

I think that now, everything works fine. At least, after having reloaded the new AS 3.13.0, I got the log file in /esarchive/ even after a failure of one single SIM job inside a wrapper (with a different experiment though). I'm not totally sure it was exactly the same case than when I opened this issue but it seems to be. 

Many thanks @dbeltrankyl for the fix. I'm closing this issue as well.
